src/components/ParserComponent.tsx

```diff
@@ -240,110 +240,125 @@ export default class ParserComponent extends React.Component<ParserInterface,
-          'inferTypes(): data (array of objects) is null');
-      assert.notStrictEqual(data, [], 'data (array of objects) is empty');
-
-      if (this.state.data.length > 0) {
-        assert.notStrictEqual(this.state.data[0], null,
-            'inferTypes(): this.state.data[0] is null');
-        const listFields = Object.keys(this.state.data[0]);
-        assert.notStrictEqual(listFields, null,
-            'inferTypes(): listFields is null');
-        assert(listFields.length > 0,
-            'inferTypes(): listFields is empty');
-        // instantiate objects to track the types of data
-        const typesForEachCol =
-            this.createTypeCountingObjects(listFields.length);
-        // check half the values to find if the data is consistent
-        [0, Math.floor(this.state.data.length / 2)].forEach((element) => {
-          const row: object = this.state.data[element];
-          assert.notStrictEqual(row, null,
-              'inferTypes(): row object is null');
-          // look at each field and categorize
-          for (let i = 0; i < listFields.length; i++) {
-            const curColTypes = typesForEachCol[i];
-            try {
-              // @ts-ignore
-              const val = row[listFields[i]];
-              assert.notStrictEqual(val, undefined,
-                  'inferTypes(): value in field is undefined');
-              if (typeof val === 'string') {
-                const date = moment(val);
-                const isValid = date.isValid();
-                if (isValid) {
-                  curColTypes['numDate'] += 1;
-                } else {
-                  throw val;
-                }
+   * goes through data, checks what the most common type of data is in the colum
+   * and returns a list of each and how many types were found
+   * @precondition: listFields is not empty
+   * @postcondition: creates a non-empty list of CountTypes
+   * @param {Array} listFields a list of the keys for each column
+   * @return {CountTypes[]} a list of how many and which kinds of types were
+   * found
+   */
+    private inferTypesHelper(listFields: any): CountTypes[] {
+      assert.notStrictEqual(listFields, null,
+          'inferTypesHelper(): listFields is null');
+      assert(listFields.length > 0,
+          'inferTypesHelper(): listFields is empty');
+      // instantiate objects to track the types of data
+      const typesForEachCol =
+          this.createTypeCountingObjects(listFields.length);
+      // check half the values to find if the data is consistent
+      [0, Math.floor(this.state.data.length / 2)].forEach((element) => {
+        const row: object = this.state.data[element];
+        assert.notStrictEqual(row, null,
+            'inferTypes(): row object is null');
+        // look at each field and categorize
+        for (let i = 0; i < listFields.length; i++) {
+          const curColTypes = typesForEachCol[i];
+          try {
+            // @ts-ignore
```

Comment:
Can you say why this was ignored? (I know why, but only because we worked on it together)

Reply:
yup